x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/NoopStep.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.xpack.core.ilm;
+
+/**
+ * Signals that an error was encountered during the execution of a policy on an index.
+ */
+public class NoopStep extends Step {
```

Comment:
As this is, I think it will fail to run because we don't really expect anything to be a "base" `Step` in the `IndexLifecycleRunner`, instead, I think we should make this extend `ClusterStateWaitStep` and have the `isConditionMet` always return that it was met.

It would be either that, or to special-case handling this step in the runner, but I would rather have the logic inside the step than add a special case.

Reply:
You're right, I tried locally by temporarily pointing to this `NoopStep` from the previous step and the ILM got stuck in it. I changed it now to a `ClusterStateWaitStep` and the process was able to complete successfully so I just updated this.